### PR TITLE
SAK-41905 - site info: Import from Site always shows complete list of…

### PR DESCRIPTION
**Pull request by mpellicer**
… sites, including hidden ones.

### CODE REVIEW

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/sakaiproject/sakai/pullRequest?prid=5252974&bid=17213015)

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/sakaiproject/sakai/pullRequest?prid=5252974&bid=17213015)

File: site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java [hunk 1]
```diff
@@ -4205,11 +4206,31 @@ private void pageOrderToolTitleIntoContext(Context context, SessionState state,
...
+		List<String> hiddenSiteIdList = new ArrayList<>();
+		List<Site> hiddenSiteList = new ArrayList<>();
+		List<Site> visibleSiteList = new ArrayList<>();
+		Preferences preferences = preferencesService.getPreferences(userDirectoryService.getCurrentUser().getId());
+		if (preferences != null) {
+			ResourceProperties properties = preferences.getProperties(PreferencesService.SITENAV_PREFS_KEY);
+			hiddenSiteIdList = (List<String>) properties.getPropertyList(PreferencesService.SITENAV_PREFS_EXCLUDE_KEY);
+		}
+
+		if (hiddenSiteIdList != null && !hiddenSiteIdList.isEmpty()) {
```
Comment (ern) [inline site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java]: Could be reduced to a one line lambda ;)

File: site-manage/site-manage-tool/tool/src/webapp/vm/sitesetup/chef_site-siteInfo-importMigrate.vm [hunk 2]
```diff
@@ -10,28 +10,63 @@
...
+						<div class="panel-body">
+							<p class="instruction">
+								tlang.getString("sitinfimp.hiddensites.info")
+							</p>
+							#foreach($hiddenSite in $hiddenSites)
+								#set ($siteIDCounter = $siteIDCounter + 1)
+								#if ($!currentSite && ($hiddenSite.Id != $!currentSite.Id))
+									<div class="radio">	
+										<label for="site_$siteIDCounter">
+											<input type="radio" name="importSites" id="site_$siteIDCounter" #if ($!importSiteList.containsKey($!hiddenSite)) checked="checked" #end value="$hiddenSite.getId()"/>
```
[the remark this answers is not in the document]
Comment (jonespm) [reply site-manage/site-manage-tool/tool/src/webapp/vm/sitesetup/chef_site-siteInfo-importMigrate.vm]: This is a autogenerated unique id

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/sakaiproject/sakai/pullRequest?prid=5252974&bid=17213015)

**Pull request merged**